bigframes/dataframe.py

```diff
@@ -840,57 +838,148 @@ def _repr_html_(self) -> str:
         if opts.repr_mode == "deferred":
             return formatter.repr_query_job(self._compute_dry_run())
 
-        # Process blob columns first, regardless of display mode
+        # Process blob columns first for non-deferred modes
+        df, blob_cols = self._process_blob_columns()
+
+        pandas_df, row_count, query_job = df._block.retrieve_repr_request_results(
+            max_results
+        )
+
+        self._set_internal_query_job(query_job)
+        column_count = len(pandas_df.columns)
+
+        return self._create_html_representation(
+            pandas_df, row_count, column_count, blob_cols
+        )
+
+    def _process_blob_columns(self) -> tuple[DataFrame, list[str]]:
+        """Process blob columns for display."""
         self._cached()
```

Comment:
This seems like the wrong function in which to call `_cached()`. `_cached()` can be very expensive (causes `a query). Seems better to just call it once if needed in `_repr_mimebundle_`.